### PR TITLE
Add full scarthgap support

### PR DESCRIPTION
**Pull request by Ruadhri17**
Previously only precompiled binaries were working and thin-edge could not be build from source due to outdated version of rust compiler. Now, once rust is supported in `meta-lts-mixins`, `meta-tedge` for Scarthgap has full support for all sub-layers. Additionally, kas builds for scarthgap were added and support for different thin-edge plugins.

TODO
* [x] uploaded artifact for mender jobs don't include the SD card image file `*sdimg.bz` (need to check if pattern is still incorrect or if the sdimg.bz image is built by default or not) @reubenmiller

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -28,8 +28,8 @@ jobs:
           - tedge-rauc
           - tedge-mender
         machine:
-          - raspberrypi
-          - raspberrypi2
+          # RaspberryPi 64 bit: 1 2
```
Comment (reubenmiller) [inline .github/workflows/build.yaml]: armv7 is actually the raspberry pi boards in 32 bit mode, here is the text taken from the [openembedded layer index site](https://layers.openembedded.org/layerindex/branch/master/machines/?q=raspberry&search=1). "RaspberryPi boards in 32 bit mode"

Though after digging a bit deeper, I'm now not so sure that `armv7` will cover Raspberry Pi 1 and 2...so we'll definitely need to test the compatibility here.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -3,10 +3,10 @@ name: build
 on:
   push:
     branches:
-      - kirkstone
+      - scarthgap
   workflow_dispatch:
   pull_request:
-    types: [ labeled ]
+    types: [labeled]
```
Comment (reubenmiller) [inline .github/workflows/build.yaml]: Seems like there is a yaml auto formatter in effect here...I would avoid the auto formatting stuff for now, and we can format all of the yaml files in a follow up PR (and do it to both kirkstone and scarthgap branches).

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -58,7 +58,7 @@ jobs:
       - uses: actions/setup-python@v5
         if: ${{ !startsWith(runner.name, 'tedge') }}
         with:
-          python-version: '3.8'
+          python-version: "3.8"
```
Comment (reubenmiller) [inline .github/workflows/build.yaml]: Revert auto formatting changes.

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -70,8 +70,8 @@ jobs:
...
-            name: ${{ matrix.image_name }}_${{ matrix.machine }}
-            path: |
-              kas/build/tmp/deploy/images/${{ matrix.machine }}/${{ matrix.image_name }}*${{ matrix.machine }}_*.${{ matrix.image_ext }}
-              kas/build/tmp/deploy/images/${{ matrix.machine }}/${{ matrix.image_name }}*${{ matrix.machine }}_*.${{ matrix.update_ext }}
-              kas/build/tmp/deploy/images/${{ matrix.machine }}/${{ matrix.image_name }}*${{ matrix.machine }}_*.${{ matrix.manifest_ext }}
+          name: ${{ matrix.image_name }}_${{ matrix.machine }}
+          path: |
+            kas/build/tmp/deploy/images/${{ matrix.machine }}/${{ matrix.image_name }}*${{ matrix.machine }}_*.${{ matrix.image_ext }}
+            kas/build/tmp/deploy/images/${{ matrix.machine }}/${{ matrix.image_name }}*${{ matrix.machine }}_*.${{ matrix.update_ext }}
+            kas/build/tmp/deploy/images/${{ matrix.machine }}/${{ matrix.image_name }}*${{ matrix.machine }}_*.${{ matrix.manifest_ext }}
```
Comment (reubenmiller) [inline .github/workflows/build.yaml]: Revert auto formatting changes.

(even though this change is very much needed, since the existing file was mixing space indentation between 2 and 4 spaces.)

File: kas/config/mender-raspberrypi.yaml [hunk 5]
```diff
@@ -12,7 +12,7 @@ local_conf_header:
     IMAGE_FSTYPES:remove = " rpi-sdimg wic.bz2 wic.bmap"
     SDIMG_ROOTFS_TYPE = "ext4"
 
-    MENDER_STORAGE_TOTAL_SIZE_MB ?= "2048"
+    MENDER_STORAGE_TOTAL_SIZE_MB ?= "2560"
```
Comment (reubenmiller) [inline kas/config/mender-raspberrypi.yaml]: What's the reason for this change? Did you get warnings about too little rootfs space when building for scarthgap?

File: kas/projects/tedge-rauc.yaml [hunk 6]
```diff
@@ -50,14 +50,9 @@ repos:
 
   meta-rauc-community:
     url: "https://github.com/thin-edge/meta-rauc-community"
-    branch: kirkstone-fork
     layers:
       meta-rauc-raspberrypi:
```
Comment (reubenmiller) [inline kas/projects/tedge-rauc.yaml]: The usage of the fork is still required. It was mainly done to provide a custom WIC file to change which partition is marked for growing (expanding on first boot). The meta-rauc-community sets the `/home/root` partition to be expanded, where as I prefer to expand the `/data` partition. Though if we can find a better way to provide a new WIC definition file that does not involve forking the repo, then by all means.

**Comment by reubenmiller** [comment]
@Ruadhri17 I've trigger a build (by adding the "ci" label to PR). It will run through building for all targets, and then we can test the functionality of them before we merge this PR.

**Comment by reubenmiller** [comment]
@Ruadhri17 All of the builds failed. A quick scan shows that there is a layer incompatibility (but it would be better to check the logs yourself).

```
2024-12-18 07:43:31 - INFO     - ERROR: Layer recipes-kernel is not compatible with the core layer which only supports these series: scarthgap (layer is compatible with kirkstone)
```

File: kas/config/mender-raspberrypi.yaml (hunk 5 above)
Comment (Ruadhri17) [reply kas/config/mender-raspberrypi.yaml]: Exactly, and it was not a warning but an error.

File: kas/projects/tedge-rauc.yaml (hunk 6 above)
Comment (Ruadhri17) [reply kas/projects/tedge-rauc.yaml]: In that case we need to create PR under the `meta-rauc-community` to add `scarthgap` as compatible layer (or create separate fork for scarthgap).

File: kas/config/mender-raspberrypi.yaml (hunk 5 above)
Comment (reubenmiller) [reply kas/config/mender-raspberrypi.yaml]: Ok, we'll have to review this new value and possibly backport it to kirkstone so that users can upgrade from kirkstone to scarthgap etc...the change in size might cause some problem during the upgrade as I don't think it supports resizing the root partition.

Comment (reubenmiller) [reply kas/config/mender-raspberrypi.yaml]: I've tested it and changing this value does break OTA updates, however it is better to break it now and increase it to 4096MB to give us a buffer for future changes. The change to 4096MB would provide a buffer of about 1.6GB for the root-fs (per A/B partition).

I've backported this setting to the kirkstone branch, and will rebase shortly (after rebasing the current scarthgap branch)

**Comment by reubenmiller** [comment]
@Ruadhri17 I rebased the scarthgap (on the upstream repo), and then rebased your PR. I had to modify the changes to the docker-ce bbappend, by renaming to docker-moby and I only removed the deprecated key field, but left the logic which controls where docker stores all of its data.
